odl/operator/tensor_ops.py

```diff
@@ -340,7 +340,12 @@ def derivative(self, vf):
         inner_vf = vf.copy()
 
         for gi in inner_vf:
-            gi /= vf_pwnorm_fac * gi ** (self.exponent - 2)
+            if self.exponent >= 2:
+                with np.errstate(invalid='ignore', divide='ignore'):
+                    gi /= vf_pwnorm_fac * gi ** (self.exponent - 2)
+                    gi = np.nan_to_num(x=gi)
```

Comment:
Second thought, this does not do anything, it simply modifies `gi` in here, not `inner_vf`, so it does not change the final result.

Could you add a test for this to make sure it really works?